src/library/scala/runtime/ScalaRunTime.scala

```diff
@@ -130,13 +130,20 @@ object ScalaRunTime {
   }
 
   def toArray[T](xs: scala.collection.Seq[T]) = {
-    val arr = new Array[AnyRef](xs.length)
-    var i = 0
-    for (x <- xs) {
-      arr(i) = x.asInstanceOf[AnyRef]
-      i += 1
+    if (xs.isEmpty) Array.emptyObjectArray
+    else {
+      object copier extends Function1[T, Unit] {
+        val arr = new Array[AnyRef](xs.length)
+        var i = 0
+
+        override def apply(x: T): Unit = {
+          arr(i) = x.asInstanceOf[AnyRef]
+          i += 1
+        }
+      }
+      xs foreach copier
+      copier.arr
```

Comment:
Using an iterator seems more natural - is there a reason not to?

Reply:
That does seem unnatural. But I'm reminded there is still no lint for `Ref` creation.

Reply:
I just wrote it is a for loop, because the original was like that. I am not sure where this is implicitly called from. I have changed it to an iterator

Regarding the Ref creation - there are 152 matches for `var .*\n[\W]*for` regex in the codebase. Some of these are in comments and false matches, but there are lots of patterns that it doesn't pick up
Most of the cases that I see in library and compiler are ref creation. 

We just did this case because we see a few 00Gb of IntRef, BooleanRef and ObjectRef and trying to squash them

Reply:
@som-snytt I'm reminded people keep asking for one. #8691